zkevm-circuits/src/copy_circuit.rs

```diff
@@ -610,30 +634,56 @@ impl<F: Field> CopyCircuitConfig<F> {
     }
 }
 
+/// Struct for test data, specifies values for related lookup tables
+#[cfg(test)]
+#[derive(Clone, Debug, Default)]
+pub struct CopyCircuitTestData {
+    pub max_txs: usize,
+    pub txs: Vec<Transaction>,
+    pub max_rws: usize,
+    pub rws: RwMap,
+    pub bytecodes: HashMap<Word, Bytecode>,
+}
+
 /// Copy Circuit
 #[derive(Clone, Debug)]
 pub struct CopyCircuit<F: Field> {
-    /// Block
-    pub block: Option<Block<F>>,
-    /// Max number of transactions supported
-    pub max_txs: usize,
+    /// Copy events
+    pub copy_events: Vec<CopyEvent>,
+    _marker: PhantomData<F>,
+    #[cfg(test)]
+    pub test_data: CopyCircuitTestData,
 }
 
 impl<F: Field> CopyCircuit<F> {
     /// Return a new CopyCircuit
-    pub fn new(max_txs: usize, block: Block<F>) -> Self {
+    pub fn new(copy_events: Vec<CopyEvent>) -> Self {
         Self {
-            block: Some(block),
-            max_txs,
+            copy_events,
+            _marker: PhantomData::default(),
+            #[cfg(test)]
+            test_data: CopyCircuitTestData::default(),
+        }
+    }
+
+    /// Return a new CopyCircuit with test data
+    #[cfg(test)]
+    pub fn new_with_test_data(copy_events: Vec<CopyEvent>, test_data: CopyCircuitTestData) -> Self {
+        Self {
+            copy_events,
+            _marker: PhantomData::default(),
+            test_data,
         }
     }
 }
 
 impl<F: Field> Default for CopyCircuit<F> {
     fn default() -> Self {
         Self {
-            block: Some(Block::default()),
-            max_txs: usize::default(),
+            copy_events: vec![],
+            _marker: PhantomData::default(),
+            #[cfg(test)]
+            test_data: CopyCircuitTestData::default(),
         }
     }
 }
```

Comment:
previously this wouldn't have worked because Option<T>::default() is None, but now every field is just set to its default value, which is what the derive implementation will do.